contracts/solidity/zaps/VaultCreationZap.sol

```diff
@@ -0,0 +1,198 @@
+   */
+
+  constructor(address _vaultFactory) {
+    vaultFactory = INFTXVaultFactory(_vaultFactory);
+  }
+
+
+  /**
+   * @notice Creates an NFTX vault, handling any desired settings and tokens.
+   * 
+   * @dev Tokens are deposited into the vault prior to fees being sent.
+   * 
+   * @param vaultData Basic information about the vault stored in `vaultInfo` struct
+   * @param vaultFeatures A numeric representation of boolean values for features on the vault
+   * @param vaultFees Fee definitions stored in a `vaultFeesConfig` struct
+   * @param eligibilityStorage Eligibility implementation, stored in a `vaultEligibilityStorage` struct
+   * @param assetTokens Tokens to be transferred to the vault in exchange for vault tokens
+   * 
+   * @return vaultId_ The numeric ID of the NFTX vault
+   */
+
+  function createVault(
+    vaultInfo calldata vaultData,
+    uint vaultFeatures,
+    vaultFeesConfig calldata vaultFees,
+    vaultEligibilityStorage calldata eligibilityStorage,
+    vaultTokens calldata assetTokens
+  ) external nonReentrant returns (uint vaultId_) {
+    // Create our vault skeleton
+    vaultId_ = vaultFactory.createVault(
+      vaultData.name,
+      vaultData.symbol,
+      vaultData.assetAddress,
+      vaultData.is1155,
+      vaultData.allowAllItems
+    );
+
+    // Build our vault interface
+    INFTXVault vault = INFTXVault(vaultFactory.vault(vaultId_));
+
+    // If we have specified vault features then update them
+    if (vaultFeatures > 0) {
+      vault.setVaultFeatures(
+        _getBoolean(vaultFeatures, 0),
+        _getBoolean(vaultFeatures, 1),
+        _getBoolean(vaultFeatures, 2),
+        _getBoolean(vaultFeatures, 3),
+        _getBoolean(vaultFeatures, 4)
+      );
+    }
+
+    // Mint and stake liquidity into the vault
+    uint length = assetTokens.assetTokenIds.length;
+    if (length > 0) {
+      if (!vaultData.is1155) {
+        for (uint i; i < length;) {
+          _transferFromERC721(vaultData.assetAddress, assetTokens.assetTokenIds[i], address(vault));
+          unchecked { ++i; }
+        }
+      } else {
+        IERC1155Upgradeable(vaultData.assetAddress).safeBatchTransferFrom(
+          msg.sender,
+          address(this),
+          assetTokens.assetTokenIds,
+          assetTokens.assetTokenAmounts,
+          ""
+        );
+      }
+
+      // We can now mint our asset tokens, giving the vault our tokens
+      vault.mintTo(assetTokens.assetTokenIds, assetTokens.assetTokenAmounts, msg.sender);
+    }
+
+    // Set our vault fees, converting our 9-decimal to 18-decimal
+    vault.setFees(
+      uint256(vaultFees.mintFee) * 10e9,
+      uint256(vaultFees.randomRedeemFee) * 10e9,
+      uint256(vaultFees.targetRedeemFee) * 10e9,
+      uint256(vaultFees.randomSwapFee) * 10e9,
+      uint256(vaultFees.targetSwapFee) * 10e9
+    );
+
+    // If we have a specified eligibility storage, add that on
+    if (eligibilityStorage.moduleIndex > 0) {
```

Comment:
This can be 0 to indicate a module... maybe you can handle this some other way?

Reply:
Ah that makes sense, can check >= and allow negative numbers passed I reckon, will check it out